Snowflake.Data/Core/HttpUtil.cs

```diff
@@ -341,7 +358,10 @@ internal RetryHandler(HttpMessageHandler innerHandler, bool disableRetry, bool f
                 p.ConnectionLimit = 20;      // Default value is 2, we need more connections for performing multiple parallel queries
 
                 TimeSpan httpTimeout = (TimeSpan)requestMessage.Properties[SFRestRequest.HTTP_REQUEST_TIMEOUT_KEY];
-                TimeSpan restTimeout = (TimeSpan)requestMessage.Properties[SFRestRequest.REST_REQUEST_TIMEOUT_KEY];
+                // Use the lower timeout value between rest timeout and max retry timeout
+                TimeSpan restTimeout = ((TimeSpan)requestMessage.Properties[SFRestRequest.REST_REQUEST_TIMEOUT_KEY]).TotalSeconds < retryTimeout ?

```

Comment:
Check 0 as unlimited I think the existng rest timeout support that as well.

Reply:
It checks for 0 in [here](https://github.com/snowflakedb/snowflake-connector-net/pull/803/files#diff-28552eb7d57bcb8b7f43eb5706654e031aaac0cf623630bbdfb8f7c9b61a93cfR483) to determine whether to limit the timeout or not. And if the timeout from `GetRestTimeout` is 0 when sending the [request](https://github.com/snowflakedb/snowflake-connector-net/blob/master/Snowflake.Data/Core/RestRequester.cs#L97), then it will also be unlimited.

However, if the retry timeout is shorter than the connection timeout, it won't be used when the request is sent since it's currently only checked in HttpUtil. The PR is now updated so the shorter timeout will be set in the properties instead so it will be used in both the request and the retry

Also added test case to cover retry timeout being shorter than connection timeout